### PR TITLE
tests: fix new tumbleweed image

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change make small fixes to make test snapd tests work in the image built on the one updated recently: https://download.opensuse.org/repositories/home:/maciek_borzecki:/Virtualization:/Appliances:/Images:/openSUSE-Tumbleweed/openSUSE_Tumbleweed/openSUSE-Tumbleweed-JeOS.x86_64-OpenStack-Cloud.qcow2

The image has several changes and dependencies installed to make snapd and snaps work.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -556,7 +556,7 @@ prepare: |
     fi
     if [[ "$SPREAD_SYSTEM" == opensuse-* ]]; then
         # refresh metadatadata
-        zypper ref
+        zypper --no-gpg-checks ref
```
Comment (bboozzoo) [inline spread.yaml:559]: This should be  `--gpg-auto-import-keys` which will do the right thing. We still want the signatures to be checked.

File: tests/lib/tools/os.query [hunk 2]
```diff
@@ -88,7 +88,7 @@ is_opensuse() {
 }
 
 is_opensuse_tumbleweed() {
-    grep -qFx 'ID="opensuse-tumbleweed"'
+    grep -qFx 'ID="opensuse-tumbleweed"' /etc/os-release
```
Comment (stolowski) [inline tests/lib/tools/os.query:91]: This looks fine but I don't is_opensuse_tumbleweed uses anywhere, so I presume it's the other change that fixes problems? To be clear, of course it's fine to land it in this PR.

File: spread.yaml (hunk 1 above)
Comment (stolowski) [inline spread.yaml:559]: I assume this is the actual workaround for the issues? Can you add a comment explaining what's the issue we are seeing with gpg checks?

**Review by bboozzoo — COMMENTED** [review]
I looked at the failures and at least all the workers managed to prepare the project data correctly, which is a great improvement over the vanilla image. We still need to install some dependencies, at least the lack of iptables is breaking some tests.

File: tests/lib/tools/os.query (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/tools/os.query:91]: Hi, thanks for the review, it is used here https://github.com/snapcore/snapd/blob/master/tests/lib/dirs.sh#L20

File: spread.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply spread.yaml:559]: sure, thanks for the review

Comment (sergiocazzolato) [reply spread.yaml]: done

**Comment by sergiocazzolato** [comment]
Still 4 tests failing:
. tests/main/interfaces-network-bind
. tests/main/manpages
. tests/main/snap-run
. tests/main/interfaces-x11-unix-socket

Fixes for those will be included in a following PR

**Comment by bboozzoo** [comment]
The failure in manpages is likely caused by `rpm.install.excludedocs = yes` in /etc/zypp/zypp.conf. We also need to add the `audit` package, otherwise the debug section attempts to run `auditctl` fail. The failure in interfaces-x11-unix-socket may be caused by AppArmor, but we'll know more once audit is include in the image..

I don't have any ideas about the snap-run test failing on strace with a segmentation fault. I suspected this may be caused by strace-static being too old for this kernel, but I cannot reproduce it locally here.

File: spread.yaml [hunk 3]
```diff
@@ -556,7 +556,8 @@ prepare: |
     fi
     if [[ "$SPREAD_SYSTEM" == opensuse-* ]]; then
         # refresh metadatadata
-        zypper ref
+        # Auto import gpg keys needed for could repository added to support google backend
```
Comment (stolowski) [inline spread.yaml:559]: Thanks for this comment.

**Review by stolowski — APPROVED** [review]
LGTM. I understand there are more issues, it's fine to address them in followups. Thanks

**Review by bboozzoo — APPROVED** [review]
LGTM, as discussed on IRC, more fixes will be done in a followup

**Pull request merged**